### PR TITLE
Provides path to lib and include dirs for homebrew distribution.

### PR DESCRIPTION
**Pull request by clebreto**
Leaves the option of having a flexible location for homebrew installation.

It is a slightly different alternative to that of #25473.

### CODE REVIEW

**Comment by clebreto** [comment]
CI fails on oraclelinux and alpine, these OSs have been added since the last successful pipeline ? 
The changes brought to the recipe should have no interference with these OSs.

I am very unfamiliar with these distributions, how to ensure that ppl is at least installed on the CI machines ?

**Comment by samoht** [comment]
Thanks for your patch! The failure seems to indicate that we are missing entries in the `depexts` field for those distributions. Would you mind trying to add some of those? It's also fine to merge just as is if @rtetley or @jmid have an opinion vs. #25473

**Comment by mseri** [comment]
`oraclelinux` does not provide libppl afaik and alpine has it on the edge repository, so I don't think we can improve it for the moment.

I'd wait to hear if there are other opinions, but this is good to merge also for me

**Comment by jmid** [comment]
LGTM too! :+1: 

If we want to get it to zero red CI lights, adding something like the following should work (not completely sure about the `alpine-3.19` part though):
```diff
+x-ci-accept-failures: [
+  "oraclelinux-8" "oraclelinux-9" "alpine-3.19"
+]
```

File: packages/conf-ppl/conf-ppl.1/opam [hunk 1]
```diff
@@ -5,6 +5,7 @@ bug-reports: "http://bugseng.com/products/ppl/bugs"
 license: "GPL-1.0-or-later"
 build: [
   ["sh" "-c"
+	"cc test.c -I${HOMEBREW_PREFIX:-/opt/homebrew}/include -L${HOMEBREW_PREFIX:-/opt/homebrew}/lib -lppl_c -lppl" {os = "macos" & os-distribution = "homebrew"}
 	"cc test.c -lppl_c -lppl" {os != "freebsd" & os != "openbsd"}
 	"cc test.c -I/usr/local/include -L/usr/local/lib -lppl_c -lppl" {os = "freebsd" | os = "openbsd"}]
 ]
```
Comment (mseri) [inline packages/conf-ppl/conf-ppl.1/opam:11]:
```suggestion
]
x-ci-accept-failures: [
  "oraclelinux-8" "oraclelinux-9" "alpine-3.19"
]
```

**Comment by mseri** [comment]
Let's do it, I don't see the situation improving on those distros anyway

**Pull request merged**